changelog.d/pr-8854

```diff
@@ -0,0 +1,10 @@
+synopsis: Add language extension ListTuplePuns
+packages: Cabal-syntax
+prs: #8854
+significance: significant
```

Comment:
Not sure it's a significant change for Cabal: we simply follow GHC here (it may be deemed significant on GHC side).